scene/3d/mesh_instance.cpp

```diff
@@ -280,6 +605,17 @@ void MeshInstance::_notification(int p_what) {
 	if (p_what == NOTIFICATION_ENTER_TREE) {
 		_resolve_skeleton_path();
 	}
+
+	if (p_what == NOTIFICATION_VISIBILITY_CHANGED) {
+		if (skin_ref.is_valid() && mesh.is_valid() && _is_software_skinning_enabled()) {
+			ERR_FAIL_COND(!skin_ref->get_skeleton_node());
+			if (is_visible()) {
```

Comment:
As far as I remember this should be `is_visible_in_tree` rather than `is_visible`.

This is a non-obvious quirk of godot. `is_visible_in_tree` does a recursive check for an invisible parent. `is_visible` only checks the node, so `is_visible` can return true for an invisible node.

Note this should only be done at this notification, not as a check in the update routine, because `is_visible_in_tree` is far more expensive than `is_visible`.